docs/modules/ROOT/pages/high_level/security_group_host_design.adoc

```diff
@@ -0,0 +1,466 @@
+4. This IP Tables implementation cannot be offloaded to hardware, except using a SoC/ARM based NIC which has capability to run the whole Linux OS
+
+[source,shell]
+------------------------------------------------------------
+   +-------------+                     +---------------+
+   |             |                     |               |
+   |    VM 1     |                     |     VM 2      |
+   |             |                     |               |
+   +---+------+--+                     +---+-------+---+
+       | eth  |                            |  eth  |
+       +--+---+                            +---+---+
+          |                                    |
+       +--+---+                            +---+---+
+       | tap  |                            |  tap  |
+ +-----+------+-----+                 +----+-------+-----+
+ |                  |                 |                  |
+ |   Linux Bridge 1 |                 |   Linux Bridge 2 |
+ |                  |                 |                  |
+ +-----+------+-----+                 +-----+------+-----+
+       | veth |                             | veth |
+       +--+---+                             +--+---+
+          |                                    |
+       +--+---+                             +--+---+
+       | veth |                             | veth |
++------+------+-----------------------------+------+---------+
+|                                                            |
+|                     OVS  Bridge  (br-int)                  |
+|                                                            |
++------------------------------------------------------------+
+------------------------------------------------------------
+
+
+=== After OpenStack Mitaka Release
+
+The OpenStack community recognized the challenge of IP Tables implementation for security group rules.
+With OVS implementation of connection tracking <<ovs-conntrack>>, many open source solutions including ovs_dpdk_firewall<<intel>> and Dragonflow<<dragonflow-security-group>> start leveraging it for OpenStack.
+It is natural for OpenStack Neutron to adopt it.
+OpenStack Mitaka introduced official OVS Firewall driver, which addressed many of the issues in the prior releases.
+But Alcor can do even better than that.
+
+
+== Design approach
+
+The Security Group design uses OVS and OpenFlow, building on the latest Neutron OVS firewall rule approach.
+It offers the following advantages compared to Neutron:
+
+. On-Host direct connection from VM port to OVS: it doesn't need extra Linux bridge, Linux devices, Linux kernel or IP Tables;
+. Direct and simplified communication between Alcor controller to Alcor control agent;
+. Delta update: Update of security group rule only require the changed rule to be push down to compute hosts;
+. Reduced OVS SG rules: Only one set of security group rules is configured in OVS to be shared with all ports using it
```

Comment:
How do we do this?

Reply:
I will update the wordings to clarify. Please use the below as a reference: Uses of conjunctive flows

https://docs.openstack.org/neutron/train/contributor/internals/openvswitch_firewall.html

Reply:
Yes, I've read this link. I mean, how do multiple port share a flow?

Reply:
Let's take the egress flows as an example:
1.table=TRANSIENT_TABLE,priority=100,in_port=ofport,actions=set_field:{ofport}->reg{REG_PORT},set_field:{local_vlan}->reg{REG_NET},resubmit(,BASE_EGRESS_TABLE)
2.table=BASE_EGRESS_TABLE,priority=65,reg_port=ofport,dl_type=ETHERTYPE_IP,in_port=ofport,dl_src=mac_addr,nw_src=ip_addr,actions=ct(table={RULES_EGRESS_TABLE},zone=NXM_NX_REG{REG_NET}[0..15])
3.table=RULES_EGRESS_TABLE,priority=70+,dl_type=ethertype,reg_port=ofport,nw_dst=dst_ip_prefix,nw_proto=protocol,tcp_src=sport,tcp_dst=dport,ct_state=+new-est,actions=resubmit(,ACCEPT_OR_INGRESS_TABLE)
4.table=ACCEPT_OR_INGRESS_TABLE,priority=90,dl_type=ETHERTYPE_IP,reg_port=ofport,ct_state=+new-est,actions=ct(commit,zone=NXM_NX_REG{REG_NET}[0..15]),resubmit(,ACCEPTED_EGRESS_TRAFFIC_TABLE)
If multiple port share flow#3, what should flows look like?

Reply:
The current design can reuse rule for conjunctive flows only, that's only used for security group rule with remote_group.